src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -587,10 +618,10 @@ private async Task InitializeAsync(bool ignoreFailures, IEnumerable<Configuratio
 
                 try
                 {
-                    Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
-                    SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
+                    Dictionary<KeyValueIdentifier, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
                     _watchedSettings = watchedSettings;
                     _mappedData = mappedData;
+                    SetData(await PrepareData(null, cancellationToken).ConfigureAwait(false));
```

Comment:
This looks wrong to me but not sure what the best way to do this is, since if I split up `PrepareData` I can't find how to do it without repeating code or needing to have an out param in a helper. I could make an overload that calls the main `PrepareData` instead?

Reply:
One option is to create a helper method - and that method directly calls PrepareData. Something like this:

```cs
Task<Dictionary<string, string>> ReorderAndPrepareData(Dictionary<KeyValueIdentifier, string> cachedInfoLogs, CancellationToken cancellationToken = default)
{
    // Reorder data

    // Call PrepareData
}

Task<Dictionary<string, string>> PrepareData(Dictionary<string, ConfigurationSetting> orderedData, CancellationToken cancellationToken = default)
{
    // No change needed
}
```

Then `InitializeAsync` can call `PrepareData` after converting `_mappedData` to a dictionary with string keys and `RefreshAsync` can call `ReorderAndPrepareData`.